chess/frontend/gui/src/modules/Game/subcomponents/PiecePromotion.jsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import { Square } from "./Square";
+import { WHITE } from "../constants";
+
+import blackQueen from "../../../cburnett/bq.svg"
+import whiteQueen from "../../../cburnett/wq.svg"
+import blackRook from "../../../cburnett/br.svg"
+import whiteRook from "../../../cburnett/wr.svg"
+import blackKnight from "../../../cburnett/bn.svg"
+import whiteKnight from "../../../cburnett/wn.svg"
+import blackBishop from "../../../cburnett/bb.svg"
+import whiteBishop from "../../../cburnett/wb.svg"
+import "../piecepromotion.css";
+
+export class PiecePromotion extends React.Component {
+
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    let { color, show, handleClick } = this.props;
+    let whitePieces = [whiteQueen, whiteRook, whiteKnight, whiteBishop];
+    let blackPieces = [blackQueen, blackRook, blackKnight, blackBishop];
+    let usePieces = (color === WHITE ? usePieces = whitePieces : usePieces = blackPieces);
```

Comment:
This seems like an overly long ternary. It looks like it boils down to
```javascript
let usePiece = usePieces = whitePieces
```

Wouldn't we want this?
```suggestion
    let usePieces = (color === WHITE ? whitePieces :  blackPieces);
```